### DISCUSSION TITLE
If a input dataset doesn't have a time type ontology, if it is still possible create a snapshot ?

### DISCUSSION THREAD

**Discussion by xdxxxdx** [Informatievlaanderen/VSDS-LDESServer4J#756] [Q&A] [answered]
Hello,

Just an question from the case:https://github.com/Informatievlaanderen/VSDS-LDESServer4J/issues/727
According to the current snapshot is created based on a `timestamp`, from what I understand, most time I guess it is `:prov:generatedAtTime`

If the input dataset has no path's data value is a time type (e.g. ^^<http://www.w3.org/2001/XMLSchema#date> ).
And that data is not going to be preprocessed to add `prov:generatedAtTime`

Is it still possible to create a snapshot of this kind of data stream?

Thanks

**Comment by Tomvbe** [comment, marked as answer]
Hi Xueying,

The server checks if members have a timestamp and versionOf in:

be.vlaanderen.informatievlaanderen.ldes.server.domain.snapshot.services.SnapshotValidPredicate.

So both are required to make a snapshot with the server else an exception will be thrown. We need both properties as snapshotting means keeping the **last version** of all the members.

The versionOf is used to determine what version a member belongs to, to ensure only one per version is snapshotted.
The timestamp is used to know which member of a version is the latest one, as we are only interested in the most recent version.

The timestamp and versionOf are extracted from members upon ingest using the timestampPath and versionOfPath configured on the specific eventstream.

The timestampPath is indeed often <http://www.w3.org/ns/prov#generatedAtTime> but can be configured to any path. This is configured on eventstream level.

If data being ingested does not contain statements with the configured timestampPath, the timestamp will not be known and snapshotting won't be possible.

Can you clarify what you mean with this sentence?
_If the input dataset has no path's data value is a time type (e.g. ^^http://www.w3.org/2001/XMLSchema#date )._

**Reply by xdxxxdx** [reply]
Hello Tom,
The answer is very clear, thank you.
